lib/numorstring/uint8orstring.go

```diff
@@ -54,10 +54,10 @@ func (i *Uint8OrString) UnmarshalJSON(b []byte) error {
 
 // MarshalJSON implements the json.Marshaller interface.
 func (i Uint8OrString) MarshalJSON() ([]byte, error) {
-	if num, err := i.NumValue(); err == nil {
-		return json.Marshal(num)
-	} else {
+	if i.Type == NumOrStringString {
```

Comment:
What's this change for?  I think @robbrockbank's version is intended to normalise `{"foo": "123"}` to `{"foo": 123}`.  Is there somewhere that you don't want that behaviour?

He says there might be some subtleties with the programatic API so probably worth touching base with him.